python/spider-py/src/spider_py/client/task.py

```diff
@@ -0,0 +1,109 @@
+"""Spider client task module."""
+
+import inspect
+from collections.abc import Callable
+from types import FunctionType, GenericAlias
+from typing import get_args, get_origin
+
+from spider_py import core
+from spider_py.client.data import Data
+from spider_py.core import TaskInput, TaskOutput, TaskOutputData, TaskOutputValue
+from spider_py.type import to_tdl_type_str
+
+
+class TaskContext:
+    """Spider task context."""
+
+    # TODO: Implement task context for use in task executor
+
+
+# Check the TaskFunction signature at runtime.
+# Enforcing static check for first argument requires the use of Protocol. However, functions, which
+# are Callable, are not considered a Protocol without explicit cast.
+TaskFunction = Callable[..., object]
+
+
+def _is_tuple(t: type | GenericAlias) -> bool:
+    """
+    :param t:
+    :return: Whether t is a tuple.
+    """
+    return get_origin(t) is tuple
+
+
+def _validate_and_convert_params(signature: inspect.Signature) -> list[TaskInput]:
+    """
+    Checks the parameters validity and convert them into core.TaskInput.
+    :param signature:
+    :return: Converted task parameters.
```

Comment:
```suggestion
    Validates the task parameters and converts them into a list of `core.TaskInput`.
    :param signature:
    :return: The converted task parameters.
```
core.TaskInput should be surrounded by `